src/main/java/Duke.java

```diff
@@ -1,10 +1,57 @@
+import java.nio.file.Paths;
+import java.util.Scanner;
+
 public class Duke {
+    private static final String DATABASE_DIRECTORY_PATH = Paths.get(System.getProperty("user.dir"),
+            "../../../", "data").toString();
+    private Parser parser;
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * */
+    public Duke(String databaseDirectoryPath) {
+        this.ui = new Ui();
+        this.storage = new Storage(databaseDirectoryPath);
+        this.parser = new Parser();
+
+        try {
+            this.tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            this.ui.showErrorMessage("Could not load saved tasks");
+            this.tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Starts the bot.
+     */
+    public void run() {
+        this.ui.greetUser();
+
+        boolean keepApplicationRunning = true;
+        Scanner sc = new Scanner(System.in);
+
+        while (keepApplicationRunning) {
+            try {
+                String userInput = this.ui.getUserInput(sc);
+                if (userInput.equals("bye")) {
```

Comment:
Is there a reason for having `keepApplicationRunning`? Perhaps you could consider simply using `sc.hasNext()` instead. On that note, you might want to rename it to `isRunning` as in accordance with the coding standard.